### PR TITLE
(RK-5) Add Basic User Scenario Integration Test

### PR DESCRIPTION
**Pull request by cowofevil**
Create an integration test for a basic r10k workflow. Segregate integration
tests from existing acceptance and spec tests. Created a folder layout for
integration tests. Created a pre-suite and library for future r10k integration
tests.

### CODE REVIEW

**Comment by elyscape** [comment]
Why are the configurations suffixed with 64mda?

**Comment by cowofevil** [comment]
@elyscape The "mda" suffix is short for "Master, Dashboard and Agent". The designation informs the tester that one of the hosts used for testing has all the above mentioned roles. We include the configs solely for local testing of integration tests and we actually run with a much larger matrix of platforms in our CI pipeline.

**Comment by zreichert** [comment]
Is there a Jira ticket to track this work?   If so could you please incorporate the ticket number as documented here https://confluence.puppetlabs.com/display/QA/Guide+for+Git+in+QA

File: integration/tests/user_scenario/basic.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+step 'Push Changes'
+git_add_commit_push(master, 'production', 'Update site.pp and add module.', git_environments_path)
+
+#Tests
+step 'Deploy "production" Environment via r10k'
+on(master, 'r10k deploy environment -v')
+
+agents.each do |agent|
+  step "Run Puppet Agent"
+  on(agent, puppet('agent', '--test'), :acceptable_exit_codes => 2) do |result|
```
Comment (zreichert) [inline integration/tests/user_scenario/basic.rb:44]: You should expand your acceptable_exit_codes to include 0.

Comment (cowofevil) [inline integration/tests/user_scenario/basic.rb:44]: @zreichert The Puppet run should always produce a notify message which will always register as a change so in this case if the Puppet run didn't return "2" we know that a failure occurred. Adding "0" wouldn't necessarily hurt anything, but would technically make the validation looser.

**Comment by cowofevil** [comment]
@zreichert I've updated the PR, can you review again?

**Pull request merged**